cxx-extras-Jammy/Dockerfile

```
@@ -1,3 +1,21 @@
 FROM ubuntu:jammy AS deploy
 
-RUN apt-get update && apt-get install -y build-essential cmake libboost-all-dev unzip curl tar sqlite
\ No newline at end of file
+# Add focal repo so that gcc/g++ 7 can be available
+RUN echo 'deb http://archive.ubuntu.com/ubuntu focal main universe' >> /etc/apt/sources.list
+
+RUN apt-get update && apt-get install -y build-essential cmake unzip curl wget tar sqlite
+
+# Use gcc/g++ 7
+RUN apt-get install -y gcc-7 g++-7
+RUN ln -sf /usr/bin/gcc-7 /usr/bin/gcc
+RUN ln -sf /usr/bin/g++-7 /usr/bin/g++
+
+# Compile Boost 1.80
+WORKDIR /
+RUN wget https://boostorg.jfrog.io/artifactory/main/release/1.80.0/source/boost_1_80_0.tar.gz
+RUN tar -xzf boost_1_80_0.tar.gz
+WORKDIR /boost_1_80_0
+RUN ./bootstrap.sh
+RUN ./b2 -j8 install
+WORKDIR /
+RUN rm -rf boost_1_80_0
```

This is ugly and I don't like it. Should be something like
```suggestion
RUN wget -qO- https://boostorg.jfrog.io/artifactory/main/release/1.80.0/source/boost_1_80_0.tar.gz | tar xz && \ 
	cd boost_* && \
	./bootstrap.sh && \
	./b2 -j8 install && \
	cd .. && \
	rm -rf boost_*
```